Make GameplayMenuOverlay translatable

17steen opened the pull request:
This allows translators to translate the pause and failed in game menus

Note: i'm doubtful that the placement and naming of the keys is correct

see: `failed_header` and `paused_header`

rogerqcify committed:
Make GameplayMenuOverlay translatable

This allows translator to translate the pause and failed in game menus

bdach commented on osu.Game/Localisation/GameplayMenuOverlayStrings.cs:
```diff
@@ -0,0 +1,49 @@
+    {
+        private const string prefix = @"osu.Game.Resources.Localisation.GameplayMenuOverlay";
+
+        /// <summary>
+        /// "Continue"
+        /// </summary>
+        public static LocalisableString Continue => new TranslatableString(getKey(@"continue"), @"Continue");
+
+        /// <summary>
+        /// "Retry"
+        /// </summary>
+        public static LocalisableString Retry => new TranslatableString(getKey(@"retry"), @"Retry");
+
+        /// <summary>
+        /// "Quit"
+        /// </summary>
+        public static LocalisableString Quit => new TranslatableString(getKey(@"quit"), @"Quit");
```
Not sure about these being here rather than common strings. Maybe better to start off with them being here and commonise as necessary?

17steen replied:
yes,  i'm really not sure this is well placed at all, i considered putting it in common but i was afraid that could be cluttering it

bdach commented on osu.Game/Localisation/GameplayMenuOverlayStrings.cs:
```diff
@@ -0,0 +1,49 @@
+        /// </summary>
+        public static LocalisableString PausedHeader => new TranslatableString(getKey(@"paused_header"), @"paused");
+
+        /// <summary>
+        /// "you're dead, try again?"
+        /// </summary>
+        public static LocalisableString FailedDescription => new TranslatableString(getKey(@"failed_description"), @"you're dead, try again?");
+
+        /// <summary>
+        /// "you're not going to do what i think you're going to do, are ya?"
+        /// </summary>
+        public static LocalisableString PausedDescription => new TranslatableString(getKey(@"paused_description"), @"you're not going to do what i think you're going to do, are ya?");
```
I'm not sure that we want to keep all of this strangely lowercase copy. Relevant: https://github.com/ppy/osu/pull/21765#issuecomment-1552378930, cc/ @Joehuu.

Joehuu replied:
I would be consistent and sentence case all the source strings and change the casing in code if deemed design appropriate.

Also if we have 2+ sentences, the [`ToSentence()`](https://github.com/ppy/osu-framework/blob/9f689c541b7c58459a63caf6487e36c82b34fd2b/osu.Framework/Localisation/CaseTransformableString.cs#L60-L68) method will not work currently as it just uppercases the first letter without regard to punctuation.

17steen replied:
Should I update code with sentence case?

Joehuu replied:
The `GameplayMenuOverlay` design is old, so it's probably fine to break the lowercase style for now, yes.

The other UI redesigns in figma seem to be transitioning out of that style, notably the song select footer.

bdach reviewed (commented):
probably fine but I want to see other reviews here

17steen committed:
Merge branch 'master' into make_pause_menu_translatable

rogerqcify committed:
Change to sentence casing

See: https://github.com/ppy/osu/pull/23640#discussion_r1202879352

rogerqcify committed:
Merge remote-tracking branch 'origin/make_pause_menu_translatable' into make_pause_menu_translatable

peppy commented on osu.Game/Localisation/GameplayMenuOverlayStrings.cs:
```diff
@@ -0,0 +1,49 @@
+        /// </summary>
+        public static LocalisableString PausedHeader => new TranslatableString(getKey(@"paused_header"), @"paused");
+
+        /// <summary>
+        /// "You're dead, try again?"
+        /// </summary>
+        public static LocalisableString FailedDescription => new TranslatableString(getKey(@"failed_description"), @"You're dead, try again?");
+
+        /// <summary>
+        /// "You're not going to do what i think you're going to do, are ya?"
+        /// </summary>
+        public static LocalisableString PausedDescription => new TranslatableString(getKey(@"paused_description"), @"You're not going to do what i think you're going to do, are ya?");
```
Might be a good time to rewrite these to sound less... bad.

I think these were taken from flyte's design mockups with no further thought, and I don't think they align with the general "attitude" of osu! UI text.

17steen replied:
is there a certain group of people that provide english text? i don't think i'm particularly good at that :P

17steen replied:
#23646 fixes this

17steen committed:
Merge branch 'master' into make_pause_menu_translatable

bdach commented:
I think I'd rather hold with this until https://github.com/ppy/osu/pull/23647 is decided on one way or another, since that pull adds new strings to localise to the overlay (and I'd rather not do this twice if possible).

17steen commented:
makes sense

rogerqcify committed:
Merge branch 'master' into make_pause_menu_translatable

# Conflicts:
#	osu.Game/Screens/Play/FailOverlay.cs
#	osu.Game/Screens/Play/GameplayMenuOverlay.cs
#	osu.Game/Screens/Play/PauseOverlay.cs

rogerqcify committed:
Adapt changes to fit master

17steen commented:
https://github.com/ppy/osu/blob/a01577cba528089335d6e52eb07d9f33357bd0d2/osu.Game/Screens/Play/GameplayMenuOverlay.cs#L213

Should this be handled as a localisable string with a format parameter? or still keep the concatenation?

rogerqcify committed:
GameplayMenuOverlay.cs: add translatable strings for `Retry count: ` and `Song progress: `

This makes the assumption that languages will prefer having the number on the right

17steen commented:
https://github.com/ppy/osu/blob/cfa128002873f19df493a114971e7bd312aa1fed/osu.Game/Screens/Play/GameplayMenuOverlay.cs#L214-L215

and 

https://github.com/ppy/osu/blob/cfa128002873f19df493a114971e7bd312aa1fed/osu.Game/Screens/Play/GameplayMenuOverlay.cs#L220-L221

peppy committed:
Merge branch 'master' into make_pause_menu_translatable

peppy committed:
Actually use paused header

bdach commented:
> Should this be handled as a localisable string with a format parameter? or still keep the concatenation?

There's no way to do compound formatting with localisable strings for now. So what you have is the best that can be done currently.

bdach reviewed (approved):
probably fine to start with
